### PR TITLE
Add test for scriptblock logging for macOS

### PR DESCRIPTION
**Pull request by TravisEz13**
## PR Summary

Add test for scriptblock logging

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: test/tools/Modules/PSSysLog/PSSysLog.psm1 [hunk 1]
```diff
@@ -406,12 +406,28 @@ class PSLogItem
...
+            }
 
             # [com.microsoft.powershell.logid]
             $splitChars = ('[', '.', ']')
 
             $item.LogId = $parts[$osLogIds.Id]
+
+            if($item.LogId -notmatch '^\[com\.microsoft\.powershell')
+            {
+                Write-Verbose "logId: $($item.LogId)" -Verbose
```
Comment (dantraMSFT) [inline test/tools/Modules/PSSysLog/PSSysLog.psm1]: Suggest changing the verbose message slightly, to something like:

"Skipping logId: $($item.LogId)"

File: test/tools/Modules/PSSysLog/PSSysLog.psm1 [hunk 2]
```diff
@@ -789,12 +805,13 @@ function Get-PSOsLog
...
     if ($TotalCount -eq 0)
     {
-        Get-Content @contentParms | ConvertFrom-OsLog -After $After -Id $Id
+        Get-Content @contentParms | Where-Object {![string]::IsNullOrEmpty($_)} | ConvertFrom-OsLog -After $After -Id $Id
     }
     else
     {
         [string] $filter = [string]::Format("com.microsoft.powershell.{0}: (", $id)
-        Get-Content @contentParms -filter {$_.Contains($filter)} | ConvertFrom-OsLog -Id $Id -After $After | Select-Object -First $maxItems
+        Write-Warning "this code path `Get-PSOsLog -TotalCount` should not be used if the message field is needed!"
```
Comment (dantraMSFT) [inline test/tools/Modules/PSSysLog/PSSysLog.psm1]: I suggest filing an issue against this.  The intent is for the entire log entry to be returned even when using TotalCount.

**Review by dantraMSFT — APPROVED** [review]
Two minor comments; otherwise LGTM.

**Review by anmenaga — APPROVED** [review]
LGTM

**Pull request merged**